src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/Maintainability/CodeMetricsComputationTests.cs

```diff
@@ -550,6 +550,29 @@ class C2 : C1 { }
             VerifyCSharp(source, expectedMetricsText);
         }
 
+        [Fact]
+        public void MethodWithAnonymousType()
+        {
+            var source = @"
+class C
+{
+    object M1(C1 c)
+    {
+        return new {a = 1};
+    }
+}
+";
+
+            var expectedMetricsText = @"
+Assembly: (Lines: 8, ExecutableLines: 1, MntIndex: 94, CycCxty: 1, CoupledTypes: {C1}, DepthInherit: 1)
```

Comment:
I presume the test would fail without your product fix, correct? CoupledTypes should have listed the anonymous type?

Reply:
Yes, I used Test-Driven Development, writing this test first and verifying that it failed as expected (listing the anonymous type). Then I got the test passing by changing the MetricsHelper.